lib/ansible/module_utils/network/meraki/meraki.py

```diff
@@ -235,8 +235,10 @@ def request(self, path, method=None, payload=None):
         self.response = info['msg']
         self.status = info['status']
 
-        if self.status >= 300:
+        if self.status >= 500:
             self.fail_json(msg='Request failed for {url}: {status} - {msg}'.format(**info))
+        elif self.status >= 300:
+            self.fail_json(msg='Request failed for {url}: {status} - {msg} - {body}'.format(**info))
```

Comment:
So IMO it's better to return that information not as a single string, but as different JSON keys. So that information is then also useful for users to register and work with.

So something like:
```python
    self.fail_json(msg='Request failed for {url}: {status} - {msg}'.format(**info), body=info['body'])
```
Not sure if your **fail_json()** covers this already though. But then there's no need to diversify here. Why not always return the body on failure ?

Reply:
That makes sense. I separated it out as I was under the impression 5xx errors don't return a body.

Reply:
Hmm, you may be right, but I would be adding it to result, rather than the msg-string.